src/Nethermind/Nethermind.Runner/Program.cs

```diff
@@ -591,4 +593,57 @@ private static string GetProductInfo()
 
         return info.ToString();
     }
+
+    private static HashSet<string> generateValidConfigOptionsHash()
+    {
+        var options = new HashSet<string>();
+
+        Type configurationType = typeof(IConfig);
+        IEnumerable<Type> configTypes = TypeDiscovery.FindNethermindBasedTypes(configurationType)
+            .Where(ct => ct.IsInterface);
+
+        foreach (Type configType in configTypes.Where(ct => !ct.IsAssignableTo(typeof(INoCategoryConfig))).OrderBy(c => c.Name))
+        {
+            if (configType is null)
+            {
+                continue;
+            }
+
+            ConfigCategoryAttribute? typeLevel = configType.GetCustomAttribute<ConfigCategoryAttribute>();
+
+            if (typeLevel is not null && (typeLevel?.DisabledForCli ?? true))
+            {
+                continue;
+            }
+
+            foreach (PropertyInfo propertyInfo in configType
+                            .GetProperties(BindingFlags.Public | BindingFlags.Instance)
+                            .OrderBy(p => p.Name))
+            {
+                ConfigItemAttribute? configItemAttribute = propertyInfo.GetCustomAttribute<ConfigItemAttribute>();
+                if (!(configItemAttribute?.DisabledForCli ?? false))
+                {
+                    options.Add($"--{configType.Name[1..].Replace("Config", string.Empty)}.{propertyInfo.Name}");
+                }
+            }
+        }
+
+        return options;
+    }
```

Comment:
Wouldn't it be better to use `CommandLineApplication.GetOptions()` to get all options and check on those instead of parsing them again here from config? And not all options are from configs, some are added explicitly!

Reply:
Hey @LukaszRozmej 
I just run a diff against what I had pushed and seems like I forgot to push the most updated version of the latest commit
I did actually noticed that some options are added explicitly and for that reason I added this
<img width="462" alt="image" src="https://github.com/user-attachments/assets/44a85e3e-e3ad-43c2-b22a-148685430099">

But not that you mention the `.GetOptions()` methods seems like a better way to get them
Gonna give it a try 
Thanks

Reply:
You were right @LukaszRozmej 
I didn't need a new method to get all the options, the way you proposed did it for me
Here's a sample of the options I printed in the console to test
<img width="804" alt="image" src="https://github.com/user-attachments/assets/216f40a7-4c09-47f8-81ab-c4063bbed0e3">

Reply:
Yep, didn't know it was an option until I looked for it either. But it makes sense it will be there.